tests/cpu/algorithms/compare_with_acts_seeding.cpp

```diff
@@ -0,0 +1,349 @@
+                n_seed_match++;
+            }
+        }
+    }
+
+    float seed_match_ratio = float(n_seed_match) / seeds.total_size();
+    EXPECT_TRUE((seed_match_ratio > 0.95) && (seed_match_ratio <= 1.));
+
+    /*--------------------------------
+      ACTS track params estimation
+      --------------------------------*/
+    const Acts::GeometryContext geoCtx;
+
+    std::vector<Acts::BoundVector> acts_params;
+
+    for (auto& outputVec : seedVector) {
+        for (auto& seed : outputVec) {
+
+            auto& spacePoints = seed.sp();
+
+            // get SpacePointPtr
+            std::array<const SpacePoint*, 3> spacePointPtrs;
+            spacePointPtrs[0] = spacePoints[0];
+            spacePointPtrs[1] = spacePoints[1];
+            spacePointPtrs[2] = spacePoints[2];
+
+            // find geometry id
+            auto spB = spacePoints[0];
+            traccc::geometry_id geo_id = 0;
+            for (std::size_t i_h = 0; i_h < spacepoints_per_event.size();
+                 i_h++) {
+                auto& items = spacepoints_per_event.get_items()[i_h];
+                if (std::find(items.begin(), items.end(), spB) != items.end()) {
+                    geo_id = spacepoints_per_event.get_headers()[i_h];
+                    break;
+                }
+            }
+
+            EXPECT_TRUE(geo_id != 0);
+
+            auto& tf3 = surface_transforms[geo_id];
+            Acts::Vector3 center;
+            center(0, 0) = tf3._data(0, 3);
+            center(1, 0) = tf3._data(1, 3);
+            center(2, 0) = tf3._data(2, 3);
+
+            Acts::Vector3 normal;
+            normal(0, 0) = tf3._data(0, 2);
+            normal(1, 0) = tf3._data(1, 2);
+            normal(2, 0) = tf3._data(2, 2);
+
+            auto bottomSurface =
+                Acts::Surface::makeShared<Acts::PlaneSurface>(center, normal);
+
+            // Test the full track parameters estimator
+            auto fullParamsOpt = estimateTrackParamsFromSeed(
+                geoCtx, spacePointPtrs.begin(), spacePointPtrs.end(),
+                *bottomSurface, Acts::Vector3(0, 0, 2), 0.1);
+
+            acts_params.push_back(*fullParamsOpt);
+        }
+    }
+
+    // params equality check
+    int n_params_match = 0;
+    for (auto& traccc_param : traccc_params) {
+        auto& traccc_vec = traccc_param.vector();
+        for (auto& acts_vec : acts_params) {
+            if (acts_vec == traccc_vec) {
+                n_params_match++;
+                break;
+            }
+        }
+    }
+    float params_match_ratio = float(n_params_match) / traccc_params.size();
+    EXPECT_TRUE((params_match_ratio > 0.95) && (params_match_ratio <= 1.));
+}
+
+// Google Test can be run manually from the main() function
+// or, it can be linked to the gtest_main library for an already
+// set-up main() function primed to accept Google Test test cases.
+int main(int argc, char** argv) {
+    ::testing::InitGoogleTest(&argc, argv);
+
+    return RUN_ALL_TESTS();
+}
```

Comment:
The test should rather be linked against `GTest::gtest_main` than to implement the `main()` function for ourselves.

Reply:
Okay

Reply:
I simply removed `int main(..)` since it was already linked to gtest. Not sure if it is what you meant though